lib/brush.h

```diff
@@ -0,0 +1,77 @@
+#ifndef BRUSH_H
+#define BRUSH_H
+
+#include <QObject>
+#include <QBitmap>
+#include <QPixmap>
+#include <QTransform>
+#include <QDebug>
+
+#include "slate-global.h"
+
+class QPainter;
+
+class SLATE_EXPORT Brush {
+    Q_GADGET
+
+public:
+    enum Type {
+        SquareType,
+        CircleType,
+        ImageType,
+    };
+    Q_ENUM(Type)
+
+    Brush() : type(), size(), pixmap(), handle() {}
+    Brush(const Brush &other) : type(other.type), size(other.size), pixmap(other.pixmap), handle(other.handle) {}
+    Brush(const Type type, const QSize &size, const QPointF handle = {0.5, 0.5}, const bool relativeHandle = true);
+    Brush(const QImage &image, const QPointF handle = {0.5, 0.5}, const bool relativeHandle = true) :
+        Brush(ImageType, image.size(), handle, relativeHandle)
+    {
+        pixmap = QPixmap::fromImage(image);
+    }
+
+    bool operator==(const Brush &other) const {
```

Comment:
If it's not too much of a hassle, please move these definitions to the cpp file.